pkg/security/ebpf/c/defs.h

```diff
@@ -262,6 +262,35 @@ struct path_key_t {
     u32 path_id;
 };
 
+struct bpf_map_def SEC("maps/path_id") path_id = {
+    .type = BPF_MAP_TYPE_ARRAY,
+    .key_size = sizeof(u32),
+    .value_size = sizeof(u32),
+    .max_entries = 1,
+    .pinning = 0,
+    .namespace = "",
+};
+
+static __attribute__((always_inline)) u32 get_path_id(int invalidate) {
+    u32 key = 0;
+
+    u32 *prev_id = bpf_map_lookup_elem(&path_id, &key);
+    if (!prev_id) {
+        u32 first_id = 1;
+        bpf_map_update_elem(&path_id, &key, &first_id, BPF_ANY);
+
+        return first_id;
+    }
+
+    u32 id = *prev_id;
```

Comment:
Took me a while to get it, but you're doing that to make sure that the dentry is still resolvable in user space for the current event (using the old path_id), but any future event requiring this dentry will have to run the entry dentry resolver (because of the new path_id), right ?

Reply:
Exactly :), basically for after unlink, rmdir and rename

Reply:
Random thought without any context: maybe a comment will help the next reader.